src/Middleware/CheckTokenAllowedAge.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelCom\AuthModule\Middleware;
+
+use Carbon\Carbon;
+use Closure;
+use Illuminate\Http\Request;
+use MyParcelCom\AuthModule\JwtRequestAuthenticator;
+use MyParcelCom\JsonApi\Exceptions\InvalidAccessTokenException;
+
+readonly class CheckTokenAllowedAge
+{
+    public function __construct(
+        private JwtRequestAuthenticator $requestAuthenticator,
```

Comment:
Since we're going through all the trouble of defining a `RequestAuthenticatorInterface` I guess we have to use this here as well
```suggestion
        private RequestAuthenticatorInterface $requestAuthenticator,
```